server-common/src/main/java/org/apache/kafka/server/metrics/KafkaYammerMetrics.java

```diff
@@ -53,16 +72,21 @@ public static MetricsRegistry defaultRegistry() {
     }
 
     private final MetricsRegistry metricsRegistry = new MetricsRegistry();
-    private final FilteringJmxReporter jmxReporter = new FilteringJmxReporter(metricsRegistry,
-        metricName -> true);
+    private FilteringJmxReporter jmxReporter;
 
     private KafkaYammerMetrics() {
-        jmxReporter.start();
```

Comment:
So previously the jmx reporter started from the point that the `KafkaYammerMetrics` class was initialized. But now it's started only once `configure` is called. In `BrokerServer` `configure()` is not called immediately on access of `KafkaYammerMetrics.INSTANCE`. It's not clear from a quick look at the code that initialization ever happens, or if it does, what might go unobserved due to the deferred start.

Reply:
The `configure()` method is called from:
- In KRaft: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaRaftServer.scala#L60
- In ZK: https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/KafkaServer.scala#L241

Reply:
Looking at this further I'm not sure this is the best way to make this work. I wonder if we should apply the same treatment to `kafka.metrics.reporters` as we're doing to `metric.reporters`. The issue is that `FilteringJmxReporter` does not implement `KafkaMetricsReporter` but instead has its own API (based on Yammer's `JmxReporter`) so this would require quite a bit of changes.

The other alternative is to not handle these Yammer reporters as the KIP only covered `JmxReporter` and `metric.reporters`.

Reply:
It was https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/server/BrokerServer.scala#L157 where it wasn't clear to me when `configure()` (and therefore now `start()`) gets called. 

Since the KIP didn't mention the Yammer reporters it's probably a good idea to not touch them here.

Reply:
Agreed, I've reverted the changes in this file.